src/BenchmarkDotNet/Toolchains/NativeAot/Generator.cs

```diff
@@ -188,9 +190,18 @@ protected override void GenerateProject(BuildPartition buildPartition, Artifacts
 </Project>";
 
         private string GetTrimmingSettings()
-            => rootAllApplicationAssemblies
-                ? "<PublishTrimmed>false</PublishTrimmed>"
-                : "<TrimMode>link</TrimMode>";
+        {
+            StringBuilder sb = new StringBuilder();
+
+            sb.AppendLine(rootAllApplicationAssemblies ? "<PublishTrimmed>false</PublishTrimmed>" : "<TrimMode>link</TrimMode>");
+
+            if (!string.IsNullOrEmpty(trimmerDefaultAction))
+            {
+                sb.Append($"<TrimmerDefaultAction>{trimmerDefaultAction}</TrimmerDefaultAction>");
+            }
+
+            return sb.ToString();
+        }
```

Comment:
Nit: these setting got aligned with IL Linker defaults, so it's now:

* PublishTrimmed is implied to be always true when publishing with NativeAOT. At least the framework needs to be trimmed because the experience of not trimming that is just really bad.
* Application assemblies are rooted by default. One can opt them into trimming with `<TrimmerDefaultAction>link</TrimmerDefaultAction>`

So one could actually fold this into the existing `bool rootAllApplicationAssemblies`: if it's true, do nothing. If it's false, add `<TrimmerDefaultAction>link</TrimmerDefaultAction>`.

Reply:
It's almost hidden, but `<TrimmerDefaultAction>link</TrimmerDefaultAction>` is our default now.

https://github.com/dotnet/BenchmarkDotNet/blob/09d9396a7980dcb65be97169b8611c208af075a2/src/BenchmarkDotNet/Toolchains/NativeAot/NativeAotToolchainBuilder.cs#L23

Reply:
I've sent https://github.com/dotnet/BenchmarkDotNet/pull/1975 to address it